### ISSUE TITLE
:Zpdfnote not doing anything

### ISSUE THREAD

**Issue by sh-cau** [jalvesaq/zotcite#59]
I am confused why this does not work. I am able to use the @<C-X><C-O> autocompletion.
However, when I try to extract the highlighted text and notes from any document, nothing happens, and I do not get any error message.
My :Zinfo output reads

> Information from the Python module:
  zotero.sqlite     : /home/sh/Zotero/zotero.sqlite
  tmpdir            : /home/sh/.cache/zotcite
  data dir          : /home/sh/Zotero
  zotero.py         : /home/sh/repos/dotfiles/.vim/plugged/zotcite/python3/zotero.py
  attachments dir   : 
  references found  : 2137
  excluded fields   : []
  citation template : {Authors}_{Year}
  docs              : {}
  banned words      : a an the some from on in to of do with

I use the zotfile plugin to manage my files, which are stored in a nextcloud folder, but all shortcuts to receive information on a paper work just fine and also show the correct path, e.g., I am able to open the corresponding pdf with  `<leader>zo`.
Any tested file has zotero notes, and highlighted text in different colors.

What am I doing wrong or what other information do you need?

**Comment by jalvesaq** [comment]
Please, try again. You should get a helpful error message now.

**Comment by sh-cau** [comment]
Thank you for the quick response. However, the issue still remains as I have installed both python3-poppler-qt5 (v0,24,2-3build7) and python3-pyqt5 (5.14.1+dfsg-3build).

**Comment by sh-cau** [comment]
I did update the plugin. It turns out that my installation of PyQt and/or especially of python-poppler-qt5 was faulty. 
However, I am unable to extract highlighted text, which is mainly what I need. 
Notes that were added with "Add Note" in Zotero do work now.

**Comment by jalvesaq** [comment]
Did you try `:Znote` instead of `:Zpdfnote`?

**Comment by jalvesaq** [comment]
The note gets imported by `:Znote` with many HTML tags. You can get a cleaner text if you drop the note into Neovim. Then, it is pasted as Markdown which is easier to manually clean.

But it would be much better to clean the note while importing it or, even better, import the annotations directly from the SQL database. I may do either of these in the future if nobody writes a pull request for it. For now, I'm reopening the issue.

**Comment by sh-cau** [comment]
That's true. I am using Vim. It is also possible with zoteros Plugin MdNotes. But I think the workflow is rather cumbersome to manually extract the highlights as a note in zotero, then converting them to markdown.
I agree that your suggestion would be very helpful to extract it directly from the SQL database.

**Comment by jalvesaq** [comment]
@sh-cau, could you please update the plugin and try `:Zannotations key`?

**Comment by sh-cau** [comment]
Beautiful, works nicely. Thank you so much!

**Comment by jalvesaq** [comment]
Now, it's possible to fix page numbers from `:Zannotations` when Zotero doesn't get them right: https://github.com/jalvesaq/zotcite/commit/913ef73ba16d5ca7e1c710515d34c36e1a8fbf6e
